public void testCallingJavaFunctions() throws Exception {
    double[] values = { 0.0, -1.0, 1.0, Double.NaN, Double.POSITIVE_INFINITY, 100.0 };
    for (double value : values) {
        assertEquals(Math.sqrt(value), evaluateDouble("java('java.lang.Math', 'double sqrt(double)', A)", value));
        assertEquals(String.valueOf(value), evaluateString("java('java.lang.String', 'String valueOf(double)', A)", value));
        assertEquals(badRandom(value, value * value, 1 / value), evaluateDouble("java('weka.core.expressionlanguage.ExpressionLanguageTest', 'double badRandom(double, double, double)', A, A^2, 1/A)", value));
    }
    assertEquals(Thread.interrupted(), evaluateBoolean("java('java.lang.Thread', 'boolean interrupted()')"));
}
